### PR TITLE
Deprecating UnbufferedBase64InputStream

### PR DESCRIPTION
**Pull request by jglick**
Complementing #5038. Seems that #4636 orphaned this class, and that is good too because [JENKINS-61452](https://issues.jenkins-ci.org/browse/JENKINS-61452) may have been caused by #2958 or #4169.

### Proposed changelog entries

* Deprecate `UnbufferedBase64InputStream`.

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Review by jvz — APPROVED** [review]
LGTM. This can be replaced with classes from Java now (Base64.Encoder/Decoder.wrap, plus a buffered stream if needed).

Edit: I see that's what the javadoc says. Carry on!

**Review by res0nance — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Review by oleg-nenashev — APPROVED** [review]

**Pull request merged**
